### PR TITLE
feat: PWA using `@vite-pwa/nuxt` module with auto update strategy

### LINKED ISSUE #53: PWA Support
### Clear and concise description of the problem

PWA Support is missing feature of this template which is very important one.

### Suggested solution

Maybe it helps 
https://github.com/kevinmarrec/nuxt-pwa-module

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by userquin**
### Description

Add PWA, same as on Vitesse repo

### Linked Issues

None

### Additional context

None

closes #53

### CODE REVIEW

File: app.vue [hunk 1]
```diff
@@ -1,16 +1,12 @@
...
-    {
-      rel: 'icon', type: 'image/png', href: '/nuxt.png',
-    },
-  ],
 })
 </script>
 
 <template>
   <NuxtLayout>
+    <VitePwaManifest />
```
Comment (userquin) [inline app.vue:9]: Mové manifest outside layout

File: nuxt.config.ts [hunk 2]
```diff
@@ -15,4 +20,104 @@ export default defineNuxtConfig({
...
+    registerType: 'autoUpdate',
+    scope: '/',
+    base: '/',
+    manifest: {
+      id: '/',
+      scope: '/',
+      name,
+      short_name: name,
+      description: name,
+      theme_color: '#ffffff',
```
Comment (antfu) [inline nuxt.config.ts:60]: Could we move those optionally (as much as possible) by providing smart defaults?

File: nuxt.config.ts [hunk 3]
```diff
@@ -15,4 +20,104 @@ export default defineNuxtConfig({
...
+          type: 'image/png',
+          purpose: 'any maskable',
+        },
+      ],
+    },
+    workbox: {
+      globPatterns: ['**/*.{js,css,html,txt,png,ico,svg}'],
+      navigateFallbackDenylist: [/^\/api\//],
+      navigateFallback: '/',
+      cleanupOutdatedCaches: true,
```
Comment (antfu) [inline nuxt.config.ts:84]: These could also go defaults

Comment (userquin) [reply nuxt.config.ts:84]: I'll create `config` folder and add pwa and constants modules: I think the pwa module should be reviewed, no makes sense to add some defaults, for example, if the user decides to use `injectManifest` strategy, `workbox` entry will not be required.

**Pull request merged**